overlord/devicestate/devicestate.go

```diff
@@ -556,7 +590,19 @@ func prereqsFromSnapTaskSet(ts *state.TaskSet) ([]string, error) {
 	return nil, fmt.Errorf("internal error: cannot identify task-snap-setup in taskset")
 }
 
-func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Model, deviceCtx snapstate.DeviceContext, fromChange string) ([]*state.TaskSet, error) {
+func sideInfoAndPathFromID(sis []*snap.SideInfo, paths []string, id string) (*snap.SideInfo, string) {
+	for i, si := range sis {
+		if si.SnapID == id {
+			return sis[i], paths[i]
+		}
+	}
+	return nil, ""
+}
+
+func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Model,
+	deviceCtx snapstate.DeviceContext, fromChange string,
+	localSnaps []*snap.SideInfo, paths []string) ([]*state.TaskSet, error) {
```

Comment:
these should come before deviceCtx and fromChange which should stay last I think